android/java/org/chromium/chrome/browser/playlist/kotlin/playback_service/VideoPlaybackService.kt

```diff
@@ -0,0 +1,297 @@
+import org.chromium.chrome.browser.playlist.kotlin.util.PlaylistPreferenceUtils.rememberFilePlaybackPosition
+import org.chromium.chrome.browser.playlist.kotlin.util.PlaylistPreferenceUtils.rememberListPlaybackPosition
+import org.chromium.chrome.browser.playlist.kotlin.util.PlaylistPreferenceUtils.setLatestPlaylistItem
+import org.chromium.chrome.browser.playlist.kotlin.util.PlaylistUtils
+import org.chromium.chrome.browser.preferences.ChromeSharedPreferences
+import org.chromium.chrome.browser.profiles.ProfileManager
+import org.chromium.mojo.bindings.ConnectionErrorHandler
+import org.chromium.mojo.system.MojoException
+import org.chromium.playlist.mojom.PlaylistItem
+import org.chromium.playlist.mojom.PlaylistService
+
+@UnstableApi
+class VideoPlaybackService :
+    MediaLibraryService(),
+    ConnectionErrorHandler,
+    MediaLibraryService.MediaLibrarySession.Callback,
+    Player.Listener {
+    private lateinit var mMediaLibrarySession: MediaLibrarySession
+    protected var mPlaylistService: PlaylistService? = null
+
+    companion object {
+        private lateinit var mPlayer: ExoPlayer
+        var currentPlaylistId: String = ""
+
+        private val mutableCurrentPlayingItem = MutableLiveData<String>()
+        val currentPlayingItem: LiveData<String>
+            get() = mutableCurrentPlayingItem
+
+        private fun setCurrentPlayingItem(currentPlayingItemId: String) {
+            mutableCurrentPlayingItem.value = currentPlayingItemId
+        }
+
+        private var mediaItemsInPlayer: ArrayList<MediaItem> = ArrayList()
+
+        fun addNewPlaylistItemModel(newPlaylistItem: PlaylistItem) {
+            if (ConstantUtils.DEFAULT_PLAYLIST == currentPlaylistId) {
+                val mediaItem =
+                    MediaItemUtil.buildMediaItem(
+                        newPlaylistItem,
+                        ConstantUtils.DEFAULT_PLAYLIST,
+                        ContextUtils.getApplicationContext()
+                            .resources
+                            .getString(
+                                R.string.playlist_play_later
+                            ),
+                    )
+                mediaItemsInPlayer.add(mediaItem)
+                mPlayer.addMediaItem(mediaItem)
+            }
+        }
+
+        fun removePlaylistItemModel(playlistItemId: String) {
+            mediaItemsInPlayer.forEachIndexed { index, mediaItem ->
+                if (mediaItem.mediaId == playlistItemId && mediaItem.mediaId != currentPlaylistId) {
+                    mPlayer.removeMediaItem(index)
+                    mediaItemsInPlayer.removeAt(index)
+                }
+            }
+        }
+
+        fun reorderPlaylistItemModel(playlistItemList: List<PlaylistItem>) {
+            mediaItemsInPlayer.forEachIndexed { oldIndex, mediaItem ->
+                val newIndex = playlistItemList.indexOfFirst { it.id == mediaItem.mediaId }
+                mPlayer.moveMediaItem(oldIndex, newIndex)
```

Comment:
we have such crash in the current nightly
```
Exception java.lang.IllegalArgumentException:
  at androidx.media3.common.util.Assertions.checkArgument (Assertions.java:40)
  at androidx.media3.exoplayer.ExoPlayerImpl.moveMediaItems (ExoPlayerImpl.java:708)
  at androidx.media3.common.BasePlayer.moveMediaItem (BasePlayer.java:77)
  at com.brave.playlist.playback_service.VideoPlaybackService$Companion.reorderPlaylistItemModel (VideoPlaybackService.kt:95)
  at com.brave.playlist.fragment.PlaylistFragment.onViewCreated$lambda$2 (PlaylistFragment.java:153)
  at android.view.View.performClick (View.java:7740)
  at android.view.View.performClickInternal (View.java:7710)
  at android.view.View.-$$Nest$mperformClickInternal
  at android.view.View$PerformClick.run (View.java:30629)
  at android.os.Handler.handleCallback (Handler.java:958)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loopOnce (Looper.java:257)
  at android.os.Looper.loop (Looper.java:368)
  at android.app.ActivityThread.main (ActivityThread.java:8839)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:572)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1049)
```
we should add a check on `newIndex` isn't out of bounds